types/c3/index.d.ts

```diff
@@ -765,31 +761,38 @@ export interface TickConfiguration {
 }
 
 export interface XTickConfiguration extends TickConfiguration {
```

Comment:
Indentation for this block should be 4 spaces like in the rest of the file.

Reply:
@iansan5653: 
I have updated the PR source branch and made the following changes:
* Indentation for the `XTickConfiguration` interface body fixed
* Added JSDoc comment to `YTickConfiguration.format`
* Made `YTickConfiguration.format` function optional and allow `number | string` return value - verified that it works and is correct

> What exactly is the compile error? Can you provide more info on that?

Here's the compile error that caused my build to break originally and made me create this PR:
```
Interface 'YTickConfiguration' incorrectly extends interface 'TickConfiguration'.
  Types of property 'format' are incompatible.
    Type '(x: number) => string' is not assignable to type 'string | ((x: number | Date) => string | number) | undefined'.
      Type '(x: number) => string' is not assignable to type '(x: number | Date) => string | number'.
        Types of parameters 'x' and 'x' are incompatible.
          Type 'number | Date' is not assignable to type 'number'.
            Type 'Date' is not assignable to type 'number'.ts(2430)
```